apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -13,42 +28,99 @@ type Properties = {
 };
 
 const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
-	const { control, errors, handleSubmit } = useAppForm<UserSignUpRequestDto>({
-		defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
-		validationSchema: userSignUpValidationSchema,
-	});
+	const [isPasswordHidden, setIsPasswordHidden] = useState<boolean>(true);
+	const [isConfirmPasswordHidden, setIsConfirmPasswordHidden] =
+		useState<boolean>(true);
+	const INPUT_ICON_SIZE = 20;
+
+	const handlePasswordIconPress = (): void => {
+		setIsPasswordHidden(!isPasswordHidden);
+	};
+
+	const handleConfirmPasswordIconPress = (): void => {
+		setIsConfirmPasswordHidden(!isConfirmPasswordHidden);
+	};
+
+	const { control, errors, handleSubmit, setError, watch } =
+		useAppForm<UserSignUpFormDto>({
+			defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
+			validationSchema: userSignUpValidationSchema,
+		});
 
 	const handleFormSubmit = useCallback((): void => {
-		void handleSubmit(onSubmit)();
-	}, [handleSubmit, onSubmit]);
+		void handleSubmit((signUpSubmissionData: UserSignUpFormDto) => {
+			const { password } = signUpSubmissionData;
+			const confirmPassword = watch(ConfirmPasswordCustomValidation.FIELD);
```

Comment:
```suggestion
			const { confirmPassword, ...userData } = signUpSubmissionData;
```
We don't need `watch`. Currently you are sending  confirmPassword to Backend, which should not be the case

Reply:
I logged my `submissionData` and there is no `confirmPassword`, so it could not go to the backend.
But I have to somehow reach the confirmPassword to perform validation inside the form component.
![sign-up_sibmission_data](https://github.com/user-attachments/assets/8b52aa11-11bc-4a18-9a5b-417f16dc6b37)

Reply:
Let's keep `watch`. If we add it to validation, we will have issues on Backend, since we don't send that field to the backend